shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -690,6 +695,11 @@ - (CGFloat)statusBarPadding {
   return CGRectIsNull(intersection) ? 0.0 : intersection.size.height;
 }
 
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+  [self onUserSettingsChanged:nil];
```

Comment:
Looking more broadly at the patch rather than just the part that was failing before: why are all these layout-based methods getting onUserSettingChanged: calls? traitCollectionDidChange: makes sense, but I don't understand why a relayout would change any settings (and especially why each relayout would need to update the settings 3 times, as this change appears to be doing.)

Reply:
Here's the guide I was following:
https://developer.apple.com/documentation/appkit/supporting_dark_mode_in_your_interface?language=objc

Reply:
That guide says that all four of those methods are called on appearance changes, and that appearance-sensitive code should be moved into one of them. That's very different than saying that you should re-run *all* appearance-sensitive code in *all* of them.

Reply:
Alright. Do you have a recommendation? That guide does not disambiguate execution paths that lead to those methods, so it's not clear to me which subset of those methods will actually account for all use-cases where platform brightness might change. Do you know which ones we want?

Reply:
The documentation as I read it is saying that all of those are called when the appearance changes. I would implement only `traitCollectionDidChange:` (which best matches the conceptual use here) until/unless you have evidence that there are cases where that isn't called.

Reply:
I removed the other messages. What about when the VC is first created?

Reply:
I added a `viewDidLoad` implementation and called `onUserSettingsChanged:nil` there to ensure we have the correct setting at the beginning.

Reply:
`viewDidAppear:` is already calling `onUserSettingsChanged:`

Reply:
Is there a risk of a timing issue there given that Flutter is running asynchronously? To me, `viewDidAppear` sounds like it might show the wrong brightness momentarily before Flutter updates..

Reply:
Given that Flutter is running asynchronously there's a risk either way; `willAppear` and `didAppear` are almost certainly in the same event loop anyway. There was a long chat thread about that originally, and I thought there was a specific plan around how to make the initial frame correct. But calling it when the engine is first created is a good start.